src/scancode/resource.py

```diff
@@ -1497,26 +1497,75 @@ def get_resource_basic_attributes(_resource_data):
             _is_file = _file_type == 'file'
             return _name, _path, _is_file
 
-        # Create root resource
-        root_name, root_path, root_is_file = get_resource_basic_attributes(root_data)
+        def create_empty_resource_data():
+            # Get fields from the base Resource class and the ScannedResource class
+            base_fields = attr.fields(Resource)
+            resource_fields = attr.fields(self.resource_class)
+            # Create dict of {field: field_default_value} for the dynamically created fields
+            resource_data = {}
```

Comment:
IMHO you want this to be an OrderedDict? Or does order not matter here?

Reply:
I don't think it matters there. We're just collecting the dict of field and values so we can add them in as keyword arguments when we create a Resource.